pkg/cli/image/imagesource/s3.go

```diff
@@ -34,76 +34,73 @@ type s3Driver struct {
 	Creds     auth.CredentialStore
 	CopyFrom  []string
 
-	repositories map[string]*s3.S3
+	repositories map[string]*s3.Client
 }
 
 type s3CredentialStore struct {
-	store     auth.CredentialStore
-	url       *url.URL
-	retrieved bool
+	store auth.CredentialStore
+	url   *url.URL
 }
 
-func (s *s3CredentialStore) IsExpired() bool { return !s.retrieved }
-
-func (s *s3CredentialStore) Retrieve() (credentials.Value, error) {
-	s.retrieved = false
+func (s *s3CredentialStore) Retrieve(ctx context.Context) (aws.Credentials, error) {
 	accessKeyID, secretAccessKey := s.store.Basic(s.url)
 	if len(accessKeyID) == 0 || len(secretAccessKey) == 0 {
-		return credentials.Value{}, fmt.Errorf("no AWS credentials located for %s", s.url)
+		return aws.Credentials{}, fmt.Errorf("no AWS credentials located for %s", s.url)
 	}
-	s.retrieved = true
 	klog.V(4).Infof("found credentials for %s", s.url)
-	return credentials.Value{
+	return aws.Credentials{
 		AccessKeyID:     accessKeyID,
 		SecretAccessKey: secretAccessKey,
-		ProviderName:    "DockerCfg",
+		Source:          "DockerCfg",
 	}, nil
 }
 
-func (d *s3Driver) newObject(server *url.URL, region string, insecure bool, securityDomain *url.URL) (*s3.S3, error) {
+func (d *s3Driver) newObject(server *url.URL, region string, insecure bool, securityDomain *url.URL) (*s3.Client, error) {
 	key := fmt.Sprintf("%s:%s:%t:%s", server, region, insecure, securityDomain)
 	s3obj, ok := d.repositories[key]
 	if ok {
 		return s3obj, nil
 	}
 
-	awsConfig := aws.NewConfig()
+	ctx := context.Background()
 
-	var creds *credentials.Credentials
-	creds = credentials.NewChainCredentials([]credentials.Provider{
-		&s3CredentialStore{store: d.Creds, url: securityDomain},
-		&credentials.EnvProvider{},
-		&credentials.SharedCredentialsProvider{},
-	})
+	configOpts := []func(*config.LoadOptions) error{
+		config.WithRegion(region),
+		config.WithCredentialsProvider(&s3CredentialStore{store: d.Creds, url: securityDomain}),
+	}
 
-	awsConfig.WithCredentials(creds)
-	awsConfig.WithRegion(region)
-	awsConfig.WithDisableSSL(insecure)
+	if d.UserAgent != "" {
+		httpClient := &http.Client{
+			Transport: transport.NewTransport(http.DefaultTransport, transport.NewHeaderRequestModifier(http.Header{http.CanonicalHeaderKey("User-Agent"): []string{d.UserAgent}})),
+		}
+		configOpts = append(configOpts, config.WithHTTPClient(httpClient))
+	}
 
 	switch {
 	case klog.V(10).Enabled():
-		awsConfig.WithLogLevel(aws.LogDebugWithHTTPBody | aws.LogDebugWithRequestErrors | aws.LogDebugWithSigning)
+		configOpts = append(configOpts, config.WithClientLogMode(aws.LogSigning|aws.LogRetries|aws.LogRequest|aws.LogResponse))
 	case klog.V(8).Enabled():
-		awsConfig.WithLogLevel(aws.LogDebugWithRequestErrors)
+		configOpts = append(configOpts, config.WithClientLogMode(aws.LogRetries|aws.LogRequest))
 	case klog.V(6).Enabled():
-		awsConfig.WithLogLevel(aws.LogDebug)
+		configOpts = append(configOpts, config.WithClientLogMode(aws.LogRetries))
```

Comment:
Does this map to the former logging easily? It seem to me v6 is usually already super debug and you would just log everything there. So shouldn't we just do, log everything on v6 or higher?

Reply:
Since log types in aws has changed, it is difficult to keep one to one mapping. But I updated the code to follow semantically same principal. It logs responseBody in verbosity 10. It logs response and request in verbosity 8. It logs request. 

Please let me know your thoughts about this?.

Reply:
Ok, thanks for the explanation.